lib/logstash/inputs/wmi.rb

```diff
@@ -47,23 +47,28 @@ def run(queue)
     @wmi = WIN32OLE.connect("winmgmts://")
     
     begin
-      @logger.debug("Executing WMI query '#{@query}'")
-      loop do
+      while !stop?
+        @logger.debug("Executing WMI query '#{@query}'")
         @wmi.ExecQuery(@query).each do |wmiobj|
           # create a single event for all properties in the collection
           event = LogStash::Event.new
           event["host"] = @host
           decorate(event)
           wmiobj.Properties_.each do |prop|
-            event[prop.name] = prop.value
+            if prop.value.is_a?(String)
+              event[prop.name] = prop.value.force_encoding(Encoding::UTF_8)
```

Comment:
Looking through this [stack overflow comment](http://stackoverflow.com/questions/10200544/ruby-1-9-force-encoding-but-check) are we sure that simply forcing the encoding will actually generate valid UTF-8? According to that, it seems like the answer is no. Maybe I'm mistaken?

It seems that we would want [String#encode](http://ruby-doc.org/core-2.2.0/String.html#method-i-encode-21) with `:replace` set to some value.